### ISSUE TITLE
[API] Support for credentialed requests

### ISSUE THREAD

**Issue by mordka** [aws-amplify/amplify-js#2107]
**Problem**
I need to make GET request to API that will set browser cookie from the response. This is handled via `XMLHttpRequest.withCredentials` [documentation here](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials) 
Example scenario: request to API Gateway endpoint that signs CDN cookie and returns `set-cookie` header that will be picked up by browser and used for CDN content authorization.

**Solution**
Extend @aws-amplify/api so that the following is possible.
```
API.get(apiName, path, {withCredentials: true})
```

**Alternatives**
There are no alternatives other than using custom http client, but it requires additional effort to sign requests, the main reason why @amplify/api is used.

**Additional context**
It doesn't have to be GET request, we should allow any request to be `withCredentials` - it's a standard web API. I'm surprised it hasn't been implemented here and nobody has pointed it so far.  
As a workaround I patched the library with a simple fix in here: https://github.com/aws-amplify/amplify-js/blob/master/packages/api/src/RestClient.ts#L69
```
 params = { 
    withCredentials: init.withCredentials,
    ....
}
```

**Comment by elorzafe** [comment]
Hi @mordka thanks for your feedback. I will mark this as feature request

**Comment by lorecrafting** [comment]
Any updates on this feature? Thanks!

**Comment by jaikathuria** [comment]
@elorzafe! I also stuck in the same problem. This is a very important requirement given we can't use cookies without this. 

I would love to contribute to add this feature.

**Comment by deepankarchopra** [comment]
@elorzafe I also need this in production ASAP. Is there any update on when we can expect this

**Comment by mordka** [comment]
@deepankarchopra you can use `patch-package` and the following patch
```
diff --git a/node_modules/@aws-amplify/api/lib/RestClient.js b/node_modules/@aws-amplify/api/lib/RestClient.js
index ccbb37e..1a98af0 100644
--- a/node_modules/@aws-amplify/api/lib/RestClient.js
+++ b/node_modules/@aws-amplify/api/lib/RestClient.js
@@ -117,6 +117,7 @@ var RestClient = /** @class */ (function () {
                         logger.debug(method + ' ' + url);
                         parsed_url = this._parseUrl(url);
                         params = {
+                            withCredentials: init.withCredentials,
                             method: method,
                             url: url,
                             host: parsed_url.host,

```

**Comment by deepankarchopra** [comment]
@mordka Thanks

**Comment by manueliglesias** [comment]
closed by #3634
